### PR TITLE
Improve empty list placeholder in generic list

### PR DESCRIPTION
**Pull request by jesusreal**
**Description**

Changes proposed in this pull request:

- Create a parametrized empty list view for plain table component in generic list

**Related issue(s)**

### CODE REVIEW

**Comment by jesusreal** [comment]
Dear reviewer, please check my comment on the task: https://github.com/kyma-project/console/issues/581

**Comment by maxmarkus** [comment]
When you have a lambda, there is no header anymore which allows to add lambdas. It should be a panel with bleed set to true and the table inside the panel-body. https://github.com/jesusreal/console/pull/62/files#diff-f598b5a7f00e5637674f1a44b1d24011

File: components/angular/generic-list/src/app/modules/list/empty-list/empty-list.component.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { Component, Input, Output, EventEmitter } from '@angular/core';
+
+@Component({
+  selector: 'y-empty-list',
+  templateUrl: './empty-list.component.html',
+  styleUrls: ['./empty-list.component.scss'],
+})
+export class EmptyListComponent {
+  @Input()
+  emptyListPlaceholderData: { title: string; body: string; button: string };
```
Comment (maxmarkus) [inline components/angular/generic-list/src/app/modules/list/empty-list/empty-list.component.ts]: shared interface is an overkill, what do you think?

File: components/angular/generic-list/src/app/modules/list/empty-list/empty-list.component.html [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+          {{ emptyListPlaceholderData.buttons.header }}
+        </button> 
+    </fd-panel-actions>
+    </fd-panel-header>
+    <fd-panel-body>
+      <p class="fd-has-margin-top-small">
+        {{ emptyListPlaceholderData.body }}
+      </p>
+      <p class="fd-has-margin-top-medium fd-has-margin-bottom-small">
+        <button fd-button *ngIf="emptyListPlaceholderData.buttons?.body" class="fd-button" (click)="onButtonClicked()">
```
Comment (maxmarkus) [inline components/angular/generic-list/src/app/modules/list/empty-list/empty-list.component.html]: .fd-button is not required, `fd-button` adds it already.

File: components/angular/generic-list/src/app/modules/list/empty-list/empty-list.component.ts (hunk 1 above)
Comment (jesusreal) [reply components/angular/generic-list/src/app/modules/list/empty-list/empty-list.component.ts]: Yes I think it is overkill, I would keep it as it is

File: components/angular/generic-list/src/app/modules/list/empty-list/empty-list.component.html (hunk 2 above)
Comment (jesusreal) [reply components/angular/generic-list/src/app/modules/list/empty-list/empty-list.component.html]: done, good catch!

**Comment by jesusreal** [comment]
After discussing the comment above we decided to separate data between header and body and make both optional to make the component as flexible as possible. Example of new data structure for testing purposes:
```
  lambdasEmptyListData = {
    header: {
      text: 'Bound Lambdas',
      actionButton: '+ Add Lambda'
    },
    body: {
      text: 'It looks like you don’t have any lambdas in your namespace yet.',
      actionButton: 'Add Lambda'
    }
  };
```

**Comment by jesusreal** [comment]
Hi @parostatkiem, The problem in the generic list example app was an outdated dependency of fundamental-ngx. Please check my last two commits, I added also an example in the app component of generic list.

File: components/angular/generic-list/package.json [hunk 3]
```diff
@@ -12,7 +12,7 @@
   "dependencies": {
     "@angular/router": "^7.2.1",
     "fiori-fundamentals": "^1.4.1",
-    "fundamental-ngx": "0.0.31",
+    "fundamental-ngx": "0.0.34",
     "fundamental-ui": "^1.2.2",
```
Comment (parostatkiem) [inline components/angular/generic-list/package.json]: I think you can remove this dependency

Comment (jesusreal) [reply components/angular/generic-list/package.json]: Great catch! Done.

**Pull request merged**